apidef/oas/operation.go

```diff
@@ -343,3 +347,144 @@ func (x *XTykAPIGateway) getOperation(operationID string) *Operation {
 
 	return operations[operationID]
 }
+
+type ValidateRequest struct {
+	Enabled           bool `bson:"enabled" json:"enabled"`
+	ErrorResponseCode int  `bson:"errorResponseCode,omitempty" json:"errorResponseCode,omitempty"`
+}
+
+func (v *ValidateRequest) Fill(meta apidef.ValidatePathMeta) {
+	v.Enabled = !meta.Disabled
+	v.ErrorResponseCode = meta.ErrorResponseCode
+}
+
+func (v *ValidateRequest) ExtractTo(meta *apidef.ValidatePathMeta) {
+	meta.Disabled = !v.Enabled
+	meta.ErrorResponseCode = v.ErrorResponseCode
+}
+
+func (s *OAS) fillValidateRequest(metas []apidef.ValidatePathMeta) {
+	for _, meta := range metas {
+		operationID := s.getOperationID(meta.Path, meta.Method)
+		tykOp := s.GetTykExtension().getOperation(operationID)
+
+		if tykOp.ValidateRequest == nil {
+			tykOp.ValidateRequest = &ValidateRequest{}
+		}
+
+		tykOp.ValidateRequest.Fill(meta)
+
+		if ShouldOmit(tykOp.ValidateRequest) {
+			tykOp.ValidateRequest = nil
+		}
+
+		var schema openapi3.Schema
+		schemaInBytes, err := json.Marshal(meta.Schema)
+		if err != nil {
+			log.WithError(err).Error("Path meta schema couldn't be marshalled")
+			return
+		}
+
+		err = schema.UnmarshalJSON(schemaInBytes)
+		if err != nil {
+			log.WithError(err).Error("Schema couldn't be unmarshalled")
+			return
+		}
+
+		operation := s.Paths[meta.Path].GetOperation(meta.Method)
+
+		requestBody := operation.RequestBody
+		if requestBody == nil {
+			requestBody = &openapi3.RequestBodyRef{
+				Value: openapi3.NewRequestBody(),
+			}
+
+			operation.RequestBody = requestBody
+		}
+
+		bodyContent := requestBody.Value.Content
+		if bodyContent == nil {
+			bodyContent = openapi3.NewContent()
+			requestBody.Value.Content = bodyContent
+		}
+
+		mediaType := bodyContent.Get("application/json")
```

Comment:
shall we define a const for `application/json` and use it  instead of string literal?